src/module.c

```diff
@@ -59,8 +59,72 @@
 #include "global_stats.h"
 
 #define CLUSTERDOWN_ERR "ERRCLUSTER Uninitialized cluster state, could not perform command"
+#define NOPERM_ERR "-NOPERM User does not have the required permissions to query the index"
 #define IS_INTERNAL_COMMAND(command) (!strncmp(command, "_", 1))
 
+#define VERIFY_ACL(ctx, idxR)                                                  \
+  do {                                                                         \
+    const char *idxName = RedisModule_StringPtrLen(idxR, NULL);                \
+    IndexLoadOptions lopts =                                                   \
+      {.nameC = idxName, .flags = INDEXSPEC_LOAD_NOCOUNTERINC};                \
+    StrongRef spec_ref = IndexSpec_LoadUnsafeEx(&lopts);                       \
+    IndexSpec *sp = StrongRef_Get(spec_ref);                                   \
+    if (!sp) {                                                                 \
+      return RedisModule_ReplyWithErrorFormat(ctx, "%s: no such index", idxName);\
+    }                                                                          \
+    if (!ACLUserMayAccessIndex(ctx, sp)) {                                     \
+      return RedisModule_ReplyWithError(ctx, NOPERM_ERR);                      \
+    }                                                                          \
+  } while(0);
+
+typedef struct {
+  const char *cmdName;
+  int indexNamePos;
+} CommandIndexNamePos;
+
+// Array containing commands names and the index of the index name in the
+// command arguments.
+static CommandIndexNamePos commandIndexPositions[] = {
```

Comment:
This requires extra maintenance if we move arguments around or add new commands and seems a bit tricky to maintain.

An easier alternative is to add to the command function an additional argument which will denote the position of the index name, or just the index name.

This will require adding three command handler but will make this table redundant.
It will also remove the need to loop inside the command to find the right context to run with.

Reply:
Well, this introduces some nasty naming.. But is more maintainable. Went for it.